core/src/backend/video/software.rs

```diff
@@ -0,0 +1,73 @@
+//! Pure software video decoding backend.
+
+use crate::backend::render::{BitmapInfo, RenderBackend};
+use crate::backend::video::{
+    EncodedFrame, Error, FrameDependency, VideoBackend, VideoStreamHandle,
+};
+use generational_arena::Arena;
+use swf::{VideoCodec, VideoDeblocking};
+
+/// A single preloaded video stream.
+pub enum VideoStream {}
+
+/// Desktop video backend.
+///
+/// TODO: Currently, this just proxies out to `ruffle_h263`, in the future it
```

Comment:
This note might be outdated/not justified - or just not necessary now after this got separated from the H.263 decoder.
And I had no problem adding a different software decoder, everything fell into its place nicely.